test/views/attachment_view_test.js

```diff
@@ -0,0 +1,52 @@
+describe('AttachmentView', function() {
+
+  it('should give a data url for arbitrary content', function() {
+    var attachment = { contentType: 'arbitrary/content' };
+    var view = new Whisper.AttachmentView({model: attachment}).render();
+    assert.match(view.$el.find("a[href]").attr("href"), /http:\/\/localhost/);
+  });
+
+  it('should have proper properties', function() {
+    var now = new Date().getTime();
+    var attachment = { contentType: 'image/png', data: 'grumpy cat' };
+    var view = new Whisper.AttachmentView({model: attachment, timestamp: now});
+
+    assert.match(view.contentType, /image/);
+    assert.match(view.fileType, /png/);
+    assert.equal(view.timestamp, now);
+    assert.isTrue(view.blob instanceof Blob);
+  });
+
+  it('shoud have correct filename format', function() {
+    var filename;
+
+    // mock out filesystem access
+    extension.windows.getViews = function () {
+      return [window];
+    };
+
+    var window = {
+      chrome: {
+        fileSystem: {
+          chooseEntry: function(data, cb) {
+            // get the filename
+            filename = data.suggestedName;
+          }
+        }
+      }
+    };
+
+    var epoch = new Date(0);
+    var attachment = {
+      contentType: 'image/png',
+      data: 'grumpy cat'
+    };
+    new Whisper.AttachmentView({
+      model: attachment,
+      timestamp: epoch
+    }).saveFile();
+
+    var expected = moment(epoch).format('YYYY-MM-DD-HHmmss');
+    assert(filename === 'signal-' + expected + '.png');
```

Comment:
Consider hardcoding the expected output here. If for some reason, moment changes/breaks its formatting, this test may not fail.